diff -Paur --no-dereference -- libogg.upstream/config.sub libogg/config.sub
--- libogg.upstream/config.sub
+++ libogg/config.sub
@@ -1342,7 +1342,7 @@
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
-	      | -aos* | -aros* \
+	      | -aos* | -aros* | -sortix* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
diff -Paur --no-dereference -- libogg.upstream/include/ogg/config_types.h.in libogg/include/ogg/config_types.h.in
--- libogg.upstream/include/ogg/config_types.h.in
+++ libogg/include/ogg/config_types.h.in
@@ -1,6 +1,9 @@
 #ifndef __CONFIG_TYPES_H__
 #define __CONFIG_TYPES_H__
 
+#if defined(__sortix__)
+#include <stdint.h>
+#else
 /* these are filled in by configure */
 #define INCLUDE_INTTYPES_H @INCLUDE_INTTYPES_H@
 #define INCLUDE_STDINT_H @INCLUDE_STDINT_H@
@@ -15,6 +18,7 @@
 #if INCLUDE_SYS_TYPES_H
 #  include <sys/types.h>
 #endif
+#endif
 
 typedef @SIZE16@ ogg_int16_t;
 typedef @USIZE16@ ogg_uint16_t;
