diff -Paur --no-dereference -- libstdc++.upstream/configure libstdc++/configure
--- libstdc++.upstream/configure
+++ libstdc++/configure
@@ -27278,8 +27278,9 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $acl_cv_rpath" >&5
 $as_echo "$acl_cv_rpath" >&6; }
   wl="$acl_cv_wl"
-  libext="$acl_cv_libext"
-  shlibext="$acl_cv_shlibext"
+  # PATCH: acl_cv_libext is never set. There's something really wrong with how
+  # this configure script was generated for Sortix libstdc++. It only works for
+  # cross-compilation and not native builds without this fix.
   hardcode_libdir_flag_spec="$acl_cv_hardcode_libdir_flag_spec"
   hardcode_libdir_separator="$acl_cv_hardcode_libdir_separator"
   hardcode_direct="$acl_cv_hardcode_direct"
