diff -Paur --no-dereference -- libexpat.upstream/conftools/config.sub libexpat/conftools/config.sub
--- libexpat.upstream/conftools/config.sub
+++ libexpat/conftools/config.sub
@@ -1326,7 +1326,7 @@
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
-	      | -aos* | -aros* \
+	      | -aos* | -aros* | -sortix* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
diff -Paur --no-dereference -- libexpat.upstream/Makefile.in libexpat/Makefile.in
--- libexpat.upstream/Makefile.in
+++ libexpat/Makefile.in
@@ -45,7 +45,7 @@
 APIHEADER = $(srcdir)/lib/expat.h $(srcdir)/lib/expat_external.h
 LIBRARY = libexpat.la
 
-DESTDIR = $(INSTALL_ROOT)
+DESTDIR ?= $(INSTALL_ROOT)
 
 default:  buildlib xmlwf/xmlwf@EXEEXT@
 
diff -Paur --no-dereference -- libexpat.upstream/xmlwf/readfilemap.c libexpat/xmlwf/readfilemap.c
--- libexpat.upstream/xmlwf/readfilemap.c
+++ libexpat/xmlwf/readfilemap.c
@@ -8,17 +8,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 
-#ifdef __WATCOMC__
-#ifndef __LINUX__
-#include <io.h>
-#else
 #include <unistd.h>
-#endif
-#endif
-
-#ifdef __BEOS__
-#include <unistd.h>
-#endif
 
 #ifndef S_ISREG
 #ifndef S_IFREG
