diff -Paur --no-dereference -- libpcre.upstream/Makefile.in libpcre/Makefile.in
--- libpcre.upstream/Makefile.in
+++ libpcre/Makefile.in
@@ -998,7 +998,6 @@
 ACLOCAL_AMFLAGS = -I m4
 dist_doc_DATA = \
   doc/pcre.txt \
-  doc/pcre-config.txt \
   doc/pcregrep.txt \
   doc/pcretest.txt \
   AUTHORS \
@@ -1014,7 +1013,6 @@
   doc/html/NON-AUTOTOOLS-BUILD.txt \
   doc/html/README.txt \
   doc/html/index.html \
-  doc/html/pcre-config.html \
   doc/html/pcre.html \
   doc/html/pcre16.html \
   doc/html/pcre32.html \
@@ -1165,7 +1163,7 @@
 # it is generated from pcre.h.in.
 nodist_include_HEADERS = pcre.h $(am__append_1)
 include_HEADERS = pcreposix.h $(am__append_2)
-bin_SCRIPTS = pcre-config
+bin_SCRIPTS = 
 @WITH_REBUILD_CHARTABLES_TRUE@dftables_SOURCES = dftables.c
 BUILT_SOURCES = pcre_chartables.c
 @WITH_PCRE8_TRUE@libpcre_la_SOURCES = \
@@ -1325,7 +1323,6 @@
 
 # Note that pcrecpp.3 is not in this list, but is included separately below.
 dist_man_MANS = \
-  doc/pcre-config.1 \
   doc/pcre.3 \
   doc/pcre16.3 \
   doc/pcre32.3 \
