diff -Paur --no-dereference -- libreadline.upstream/configure libreadline/configure
--- libreadline.upstream/configure
+++ libreadline/configure
@@ -2628,8 +2628,6 @@
     i[3456]86-*-beos*)
         cross_cache=${srcdir}/cross-build/x86-beos.cache
         ;;
-    *)  echo "configure: cross-compiling for $host is not supported" >&2
-        ;;
     esac
     if test -n "${cross_cache}" && test -r "${cross_cache}"; then
         echo "loading cross-build cache file ${cross_cache}"
@@ -5412,7 +5410,7 @@
   if test "$cross_compiling" = yes; then :
   { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: cannot check for sigsetjmp/siglongjmp if cross-compiling -- defaulting to missing" >&5
 $as_echo "$as_me: WARNING: cannot check for sigsetjmp/siglongjmp if cross-compiling -- defaulting to missing" >&2;}
-     bash_cv_func_sigsetjmp=missing
+     bash_cv_func_sigsetjmp=yes
 
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
diff -Paur --no-dereference -- libreadline.upstream/Makefile.in libreadline/Makefile.in
--- libreadline.upstream/Makefile.in
+++ libreadline/Makefile.in
@@ -67,9 +67,6 @@
 
 man3dir = $(mandir)/man3
 
-# Support an alternate destination root directory for package building
-DESTDIR =
-
 # Programs to make tags files.
 ETAGS = etags
 CTAGS = ctags -w
