diff -Paur --no-dereference -- libfreetype.upstream/builds/unix/configure libfreetype/builds/unix/configure
--- libfreetype.upstream/builds/unix/configure
+++ libfreetype/builds/unix/configure
@@ -14377,21 +14377,8 @@
       libpng_libsstaticconf="$LIBPNG_LIBS"
       have_libpng="yes (LIBPNG_CFLAGS and LIBPNG_LIBS)"
     else
-      # fall back to config script
-      { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for libpng-config" >&5
-printf %s "checking for libpng-config... " >&6; }
-      if which libpng-config > /dev/null 2>&1; then
-        LIBPNG_CFLAGS=`libpng-config --cflags`
-        LIBPNG_LIBS=`libpng-config --ldflags`
-        libpng_libspriv=`libpng-config --static --ldflags`
-        libpng_libsstaticconf="$libpng_libspriv"
-        have_libpng="yes (libpng-config)"
-        { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-printf "%s\n" "yes" >&6; }
-      else
-        { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
-printf "%s\n" "no" >&6; }
-      fi
+      # PATCH: libpng-config should never be used.
+      :
     fi
   fi
 fi
